pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -173,7 +173,6 @@ def build_model(
             The PyMC model object containing all the defined stochastic and deterministic variables.
         """
         model_config = self.model_config
-        self._generate_and_preprocess_model_data(X, y)
```

Comment:
@michaelraczycki There are some tests failing, and I think is because of this line. 

I think the problem is that we are doing the preprocessing twice:

- https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/model_builder.py#L463

- https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/mmm/delayed_saturated_mmm.py#L176

Is the design to preprocess the data in the build_model or fit? (at the moment is happening at both in `main` branch)

Reply:
IMO, it should be in the `build_model`; otherwise, when we run prior_predictive, we will not have the transformed data.